src/Nest/Document/Multiple/Reindex/ReindexRequest.cs

```diff
@@ -77,49 +88,65 @@ public class ReindexRequest<T> : IReindexRequest<T> where T : class
 		/// <inheritdoc/>
 		/// <param name="scrollSource">The scroll operation yielding the source documents for the reindex operation</param>
 		/// <param name="bulkAllTarget">A factory that instantiates the bulk all operation over the lazy stream of search result hits</param>
-		public ReindexRequest(IScrollAllRequest scrollSource, Func<IEnumerable<IHit<T>>, IBulkAllRequest<IHit<T>>> bulkAllTarget)
+		public ReindexRequest(IScrollAllRequest scrollSource, Func<TSource, TTarget> map, Func<IEnumerable<IHitMetadata<TTarget>>, IBulkAllRequest<IHitMetadata<TTarget>>> bulkAllTarget)
 		{
 			scrollSource.ThrowIfNull(nameof(scrollSource), "without a scroll all request we don't know where to read from!");
```

Comment:
I think we should remove the subject pronouns in the exception messages here as well